packages/injected/src/ariaSnapshot.ts

```diff
@@ -185,6 +187,31 @@ function toAriaNode(element: Element): AriaNode | null {
   return result;
 }
 
+function normalizeGenericRoles(rootA11yNode: AriaNode) {
+  const visit = (ariaNode: AriaNode) => {
+    let newChildren: (AriaNode | string)[] = [];
+    for (const child of ariaNode.children) {
+      if (typeof child === 'string') {
+        newChildren.push(child);
+        continue;
+      }
+      const isEmptyGeneric = child.role === 'generic' && child.children.length === 0;
+      const isSingleGenericChild = child.role === 'generic' && child.children.length === 1;
+      if (isSingleGenericChild) {
+        const newChild = child.children[0];
+        newChildren.push(newChild);
+        if (typeof newChild !== 'string')
+          visit(newChild);
+      } else if (!isEmptyGeneric) {
```

Comment:
a comment that text is always a child would be helpful

Reply:
Done